src/Umbraco.Core/IO/FileSystems.cs

```diff
@@ -38,92 +40,137 @@ public class FileSystems : IFileSystems
         #region Constructor
 
         // DI wants a public ctor
-        public FileSystems(IServiceProvider container, ILogger<FileSystems> logger, ILoggerFactory loggerFactory, IIOHelper ioHelper, IOptions<GlobalSettings> globalSettings, IHostingEnvironment hostingEnvironment)
+        public FileSystems(
+            ILoggerFactory loggerFactory,
+            IIOHelper ioHelper,
+            IOptions<GlobalSettings> globalSettings,
+            IHostingEnvironment hostingEnvironment,
+            IFileSystem stylesheetFileSystem)
         {
-            _container = container;
-            _logger = logger;
+            _logger = loggerFactory.CreateLogger<FileSystems>();
             _loggerFactory = loggerFactory;
             _ioHelper = ioHelper;
             _globalSettings = globalSettings.Value;
             _hostingEnvironment = hostingEnvironment;
+
+            _stylesheetsFileSystem = (ShadowWrapper)CreateShadowWrapper(stylesheetFileSystem, "css");
```

Comment:
Instead of casting this each time wouldn't it be better to have a private method like CreateShadowWrapperInternal that actually just returns ShadowWrapper and CreateShadowWrapper just calls CreateShadowWrapperInternal?

Reply:
Yeah that's a great idea